docs/anti-patterns.mdx

```diff
@@ -4,35 +4,152 @@ title: Cadence Anti-Patterns
 
 This is an opinionated list of issues that can be improved if they are found in Cadence code intended for production.
 
-## Security and Robustness
+# Security and Robustness
 
-### Events From Resources Might Not Be Unique
+## Avoid using `AuthAccount` as a function parameter
 
-A public function on a contract that creates a resource can be called by any account.
+### Problem
+
+Some may choose to authenticate or perform operations for their users by using their account addresses.
+In order to do this, a commonly seen case would be to pass the user's `AuthAccount` object
+as a parameter to a contract function to use for querying the account or storing objects directly.
+This is problematic, as the `AuthAccount` object allows access to ALL private areas of the account,
+which provides the opportunity for bad actors to take advantage of.
+
+Example:
+
+```cadence
+...
+// Imagine this code is in a contract that uses AuthAccount to authenticate users
+// To transfer NFTs
+
+// They could deploy the contract with an Ethereum-style access control list functionality
+
+pub fun transferNFT(id: UInt64, owner: AuthAccount) {
+    assert(owner(id) == owner.address)
+
+    transfer(id)
+}
+
+// But they could upgrade the function to have the same signature
+// so it looks like it is doing the same thing, but they could also drain a little bit
+// of FLOW from the user's vault, a totally separate piece of the account that 
+// should not be accessible in this function
+// BAD
+
+pub fun transferNFT(id: UInt64, owner: AuthAccount) {
+    assert(owner(id) == owner.address)
+
+    transfer(id)
+
+    // Sneakily borrow a reference to the user's Flow Token Vault
+    // and withdraw a bit of FLOW
+    // BAD
```

Comment:
Everywhere with the BAD/GOOD annotations: Name GOOD/BAD first, then describe, e.g `BAD: Sneakily ...`

Reply:
Good point. I'll update all of them